typescript/sdk/src/hook/RadixHookReader.ts

```diff
@@ -0,0 +1,107 @@
+import { RadixSDK, RadixSigningSDK } from '@hyperlane-xyz/radix-sdk';
+import { Address, WithAddress, assert, rootLogger } from '@hyperlane-xyz/utils';
+
+import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
+
+import {
+  DerivedHookConfig,
+  HookType,
+  IgpHookConfig,
+  MerkleTreeHookConfig,
+} from './types.js';
+
+export class RadixHookReader {
+  protected readonly logger = rootLogger.child({
+    module: 'RadixHookReader',
+  });
+
+  constructor(
+    protected readonly metadataManager: ChainMetadataManager,
+    protected readonly sdk: RadixSDK | RadixSigningSDK,
+  ) {}
+
+  async deriveHookConfig(address: Address): Promise<DerivedHookConfig> {
+    try {
+      if (await this.isIgpHook(address)) {
+        return this.deriveIgpConfig(address);
+      } else if (await this.isMerkleTreeHook(address)) {
+        return this.deriveMerkleTreeConfig(address);
+      } else {
+        throw new Error(`Unsupported hook type for address: ${address}`);
+      }
+    } catch (error) {
+      this.logger.error(`Failed to derive Hook config for ${address}`, error);
+      throw error;
+    }
+  }
+
+  private async deriveIgpConfig(
+    address: Address,
+  ): Promise<WithAddress<IgpHookConfig>> {
+    const igp = await this.sdk.query.core.getIgpHook({ hook: address });
+
+    assert(igp, `IGP not found for address ${address}`);
+
+    const overhead: IgpHookConfig['overhead'] = {};
+    const oracleConfig: IgpHookConfig['oracleConfig'] = {};
+
+    Object.keys(igp.destination_gas_configs).forEach((remoteDomain) => {
+      const { name, nativeToken } =
+        this.metadataManager.getChainMetadata(remoteDomain);
+
+      const gasConfig = igp.destination_gas_configs[remoteDomain];
+
+      overhead[name] = parseInt(gasConfig.gas_overhead);
+      oracleConfig[name] = {
+        gasPrice: gasConfig.gas_oracle?.gas_price ?? '',
+        tokenExchangeRate: gasConfig.gas_oracle?.token_exchange_rate ?? '',
+        tokenDecimals: nativeToken?.decimals,
+      };
```

Comment:
if a config is not set for a given domain i'd rather not include it in the final output instead of leaving its fields as an empty string.